lib/sda/host.go

```diff
@@ -381,6 +387,20 @@ func (h *Host) processMessages() {
 	}
 }
 
+func (h *Host) processRequest(e *network.Entity, r *Request) {
+	// check if the target service is indeed existing
+	var s Service
+	var ok bool
+	if s, ok = h.serviceStore.serviceByID(r.Service); !ok {
```

Comment:
I need the `s Service` variable after. If it's put in the `if` then its scope is binded to the `if` and I can't call `s.ProcessRequest` after.